### PR TITLE
[3006.x] Add log level insecure warning to more places

### PR DESCRIPTION
**Pull request by Ch3LL**
We already log this if the user is using an insecure log level: `[WARNING ] Insecure logging configuration detected! Sensitive data may be logged`

And we have it documented in every config file:

```
 # The following log levels are considered INSECURE and may log sensitive data:
 # ['garbage', 'trace', 'debug']
```

And the documentation here: https://docs.saltproject.io/en/latest/ref/configuration/examples.html

This PR makes this warning more noticeable across multiple places in the docs and when the user runs `--help`

### CODE REVIEW

**Review by whytewolf — DISMISSED** [review]

File: salt/utils/parsers.py [hunk 1]
```diff
@@ -610,9 +610,10 @@ def _mixin_setup(self):
             *self._console_log_level_cli_flags,
             dest=self._loglevel_config_setting_name_,
             choices=list(salt._logging.LOG_LEVELS),
-            help="Console logging log level. One of {}. Default: '{}'.".format(
+            help="Console logging log level. One of {}. Default: '{}'. \n The following log levels are INSECURE and may log sensitive data: {}".format(
                 ", ".join(["'{}'".format(n) for n in salt._logging.SORTED_LEVEL_NAMES]),
                 self._default_logging_level_,
+                ", ".join(insecure_log()),
```
Comment (s0undt3ch) [inline salt/utils/parsers.py:616]:
```suggestion
            help=(
                "Console logging log level. One of {}. Default: '{}'.\n"
                "The following log levels are INSECURE and may log sensitive data: {}"
            ).format(
                ", ".join(["'{}'".format(n) for n in salt._logging.SORTED_LEVEL_NAMES]),
                self._default_logging_level_,
                ", ".join(insecure_log()),
```
Minor formatting change for smaller line length, but, is the added line break actually respected?

File: salt/utils/parsers.py [hunk 2]
```diff
@@ -636,9 +637,10 @@ def _logfile_callback(option, opt, value, parser, *args, **kwargs):
             "--log-file-level",
             dest=self._logfile_loglevel_config_setting_name_,
             choices=list(salt._logging.SORTED_LEVEL_NAMES),
-            help="Logfile logging log level. One of {}. Default: '{}'.".format(
+            help="Logfile logging log level. One of {}. Default: '{}'. \n The following log levels are INSECURE and may log sensitive data: {}".format(
                 ", ".join(["'{}'".format(n) for n in salt._logging.SORTED_LEVEL_NAMES]),
                 self._default_logging_level_,
+                ", ".join(insecure_log()),
```
Comment (s0undt3ch) [inline salt/utils/parsers.py:643]: Same as above

File: tests/pytests/unit/utils/verify/test_clean_path_link.py [hunk 3]
```diff
@@ -0,0 +1,59 @@
...
+        csl.restype = ctypes.c_ubyte
+        __CSL = csl
+    flags = 0
+    if source is not None and source.is_dir():
+        flags = 1
+    if __CSL(link_name, source, flags) == 0:
+        raise ctypes.WinError()
+
+
+@pytest.fixture()
```
Comment (s0undt3ch) [inline tests/pytests/unit/utils/verify/test_clean_path_link.py:31]:
```suggestion
@pytest.fixture
```

File: tests/pytests/unit/utils/verify/test_clean_path_link.py [hunk 4]
```diff
@@ -0,0 +1,59 @@
...
+    if __CSL is None:
+        csl = ctypes.windll.kernel32.CreateSymbolicLinkW
+        csl.argtypes = (ctypes.c_wchar_p, ctypes.c_wchar_p, ctypes.c_uint32)
+        csl.restype = ctypes.c_ubyte
+        __CSL = csl
+    flags = 0
+    if source is not None and source.is_dir():
+        flags = 1
+    if __CSL(link_name, source, flags) == 0:
+        raise ctypes.WinError()
```
Comment (s0undt3ch) [inline tests/pytests/unit/utils/verify/test_clean_path_link.py:28]:
```suggestion

class Symlink:
    """
    symlink(source, link_name) Creates a symbolic link pointing to source named
    link_name
    """
    def __init__(self):
        self._csl = None
        
    def __call__(self, source, link_name):
        if self._csl is None:
            self._csl = ctypes.windll.kernel32.CreateSymbolicLinkW
            self._csl.argtypes = (ctypes.c_wchar_p, ctypes.c_wchar_p, ctypes.c_uint32)
            self._csl.restype = ctypes.c_ubyte

        flags = 0
        if source is not None and source.is_dir():
            flags = 1
            
        if self._csl(link_name, source, flags) == 0:
            raise ctypes.WinError()

@pytest.fixture(scope="module")
def symlink():
    return Symlink()
```

There, no more globals usage.

File: tests/pytests/unit/utils/verify/test_clean_path_link.py [hunk 5]
```diff
@@ -0,0 +1,59 @@
...
+    from_path = tmp_path / "linkfrom"
+    if salt.utils.platform.is_windows():
+        kwargs = {}
+    else:
+        kwargs = {"target_is_directory": True}
+    if salt.utils.platform.is_windows():
+        symlink(to_path, from_path, **kwargs)
+    else:
+        from_path.symlink_to(to_path, **kwargs)
+    return to_path, from_path
```
Comment (s0undt3ch) [inline tests/pytests/unit/utils/verify/test_clean_path_link.py:43]:
```suggestion
def setup_links(tmp_path, symlink):
    to_path = tmp_path / "linkto"
    from_path = tmp_path / "linkfrom"
    if salt.utils.platform.is_windows():
        kwargs = {}
    else:
        kwargs = {"target_is_directory": True}
    if salt.utils.platform.is_windows():
        symlink(to_path, from_path, **kwargs)
    else:
        from_path.symlink_to(to_path, **kwargs)
    return to_path, from_path
```

**Review by s0undt3ch — CHANGES_REQUESTED** [review]

File: salt/utils/parsers.py (hunk 1 above)
Comment (s0undt3ch) [reply salt/utils/parsers.py:616]: > So yes, its respected. It looks cleaner to me to have that line break, but its not required if you want to remove it.

If it's respected, leave it.

**Review by s0undt3ch — DISMISSED** [review]

**Review by whytewolf — APPROVED** [review]

**Pull request merged**